datagenerator/src/main/java/com/linkedin/restli/datagenerator/csharp/CSharpRythmGenerator.java

```diff
@@ -0,0 +1,173 @@
+ */
+public class CSharpRythmGenerator {
+  public static final String TEMPLATE_PATH_ROOT = "rythm";
+  private static final Options _options = new Options();
+  private static final Logger LOG = LoggerFactory.getLogger(CSharpRythmGenerator.class);
+  static {
+    _options.addOption(Option.builder().longOpt("resolverPath").hasArg().build());
+    _options.addOption(Option.builder().longOpt("output").required().hasArg().build());
+    _options.addOption(Option.builder().longOpt("source").required().hasArgs().build());
+  }
+
+  protected final CommandLine _cl;
+
+  public static void main(String[] args) throws IOException {
+    CommandLine cl = null;
+    try
+    {
+      final CommandLineParser parser = new DefaultParser();
+      cl = parser.parse(_options, args);
+    }
+    catch (ParseException e)
+    {
+      LOG.error("Invalid arguments: " + e.getMessage());
+      reportInvalidArguments();
+    }
+
+    final CSharpRythmGenerator generator = new CSharpRythmGenerator(cl);
+    generator.setupRythmEngine();
+    generator.generate();
+
+    Rythm.shutdown();
+  }
+
+  public CSharpRythmGenerator(CommandLine cl) {
+    _cl = cl;
+  }
+
+  public void setupRythmEngine() {
+    final Map<String, Object> config = new HashMap<>();
+    ResourceStreamResourceLoader loader = new ResourceStreamResourceLoader();
+    config.put(RythmConfigurationKey.CODEGEN_COMPACT_ENABLED.getKey(), false);
+    config.put(RythmConfigurationKey.RESOURCE_LOADER_IMPLS.getKey(), loader);
+    Rythm.init(config);
+    loader.setEngine(Rythm.engine());
+    //Rythm.engine().registerTransformer(ObjCRythmTransformer.class);
+  }
+
+  public void generate() throws IOException {
+    final File outputDirectory = new File(_cl.getOptionValue("output"));
+
+    final DataSchemaParser schemaParser = new DataSchemaParser(_cl.getOptionValue("resolverPath"));
+    final DataSchemaParser.ParseResult parseResult = schemaParser.parseSources(_cl.getOptionValues("source"));
+
+    final TemplateSpecGenerator specGenerator = new TemplateSpecGenerator(schemaParser.getSchemaResolver());
+    generateTemplateSpecs(specGenerator, parseResult);
+
+
+    for (ClassTemplateSpec spec : specGenerator.getGeneratedSpecs()) {
+      System.out.println(spec.getSchema().toString()); //TODO DEBUG
+    }
+
+    //TODO TEST
+
+    generateResultFile(specGenerator.getGeneratedSpecs(), outputDirectory);
+  }
+
+  protected void generateTemplateSpecs(TemplateSpecGenerator specGenerator, DataSchemaParser.ParseResult parseResult) {
+    parseResult.getSchemaAndLocations().forEach(specGenerator::generate);
+  }
+
+  protected void generateResultFile(Collection<ClassTemplateSpec> specs, File outputDirectory) {
+
+    for (ClassTemplateSpec spec : specs) {
+
+      if (spec instanceof EnumTemplateSpec) {
+        EnumDataSchema schema = (EnumDataSchema) spec.getSchema();
+        Map<String, Object> params = new HashMap<String, Object>();
+        params.put("className", schema.getName());
```

Comment:
Here we can simplify our rythm parameters by just passing a variable of EnumTemplateSpec or EnumDataSchema.